src/organization/organization.service.ts

```diff
@@ -299,4 +300,58 @@ export class OrganizationService {
     };
     return json?.Data?.[0]?.['subOrgs'] || [];
   }
+
+  private async getOrgDefinitionByUid(uid: string) {
+    const query = `
+    {
+      definitions(func: uid(${uid})) @filter(type(OrgDefinition))
+      ${roleDefinitionFullQuery}
+    }
+    `;
+    const res = await this.dgraph.query(query);
+    const { definitions } = res.getJson() as {
+      definitions: OrganizationDefinitionDTO[];
+    };
+    const definition = definitions[0];
+    if (!definition) {
+      return {
+        uid,
+      } as OrganizationDefinitionDTO;
+    }
+    return definition;
+  }
+
+  private getDefinitionForOrgs(orgs?: Organization[]) {
+    return (
+      orgs &&
+      Promise.all(
+        orgs.map(async org => ({
+          ...org,
+          definition: await this.getOrgDefinitionByUid(org.definition.uid),
+          subOrgs: await this.getDefinitionForOrgs(org.subOrgs),
+        })),
+      )
+    );
+  }
+
+  public async getOrganizationNestedSubOrgs(namespace: string) {
+    const res = await this.dgraph.query(`
+    {
+      orgs(func: eq(namespace, "${namespace}")) @filter(type(Org)) @recurse(depth: 20, loop: true) {
```

Comment:
Cool, I didn't know about this `@recurse` functionality. Why the need to loop? I would think there wouldn't be loops in the organization hierarchies

Reply:
If the loop would be `false` the dgraph will stop looking when found the same node as earlier :) so this is just to be sure to find all nested suborgs in whatever configuration :)